overlord/snapstate/snapmgr.go

```diff
@@ -526,6 +527,61 @@ func (m *SnapManager) ensureRefreshes() error {
 	return nil
 }
 
+// ensureForceDevmodeDropsDevmodeFromState undoes the froced devmode
+// in snapstate for forced devmode distros.
+func (m *SnapManager) ensureForceDevmodeDropsDevmodeFromState() error {
+	if !release.ReleaseInfo.ForceDevMode() {
+		return nil
+	}
+
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	// int because we might want to come back and do a second pass at cleanup
+	var fixed int
+	if err := m.state.Get("fix-forced-devmode", &fixed); err != nil && err != state.ErrNoState {
+		return err
+	}
+
+	if fixed > 0 {
+		return nil
+	}
+
+	var snaps map[string]*json.RawMessage
+	if err := m.state.Get("snaps", &snaps); err != nil {
+		if err == state.ErrNoState {
+			// no snaps -> no problem
+			goto done
+		}
+		return err
+	}
+
+	for _, name := range []string{"core", "ubuntu-core"} {
+		var snapst *SnapState
+		if snaps[name] == nil {
+			continue
+		}
+		if err := json.Unmarshal([]byte(*snaps[name]), &snapst); err != nil {
```

Comment:
Pardon my ignorance but could we use `Get(m.state, name, &snapst)` here instead of re-implementing it? This might allow us to get rid of the above getting of all snaps (or we could still do that for efficiency and use `All()`)? Or is there a reason to not do that?